src/server/game/Movement/MotionMaster.h

```diff
@@ -196,6 +198,10 @@ class TC_GAME_API MotionMaster //: private std::stack<MovementGenerator *>
         void MoveCirclePath(float x, float y, float z, float radius, bool clockwise, uint8 stepCount);
         void MoveSmoothPath(uint32 pointId, G3D::Vector3 const* pathPoints, size_t pathSize, bool walk);
         void MoveSmoothPath(uint32 pointId, Movement::PointsArray const& points, bool walk);
+        // Walk along spline chain stored in DB (script_spline_chain_meta and script_spline_chain_waypoints)
+        void MoveAlongSplineChain(uint32 pointId, uint32 dbChainId, bool walk);
+        void MoveAlongSplineChain(uint32 pointId, SplineChain const& chain, bool walk);
```

Comment:
On one hand, yes - on the other hand, we can't (easily) parse that from sniffs, while current spline data is fully sniff parsable.

Also, I haven't seen any case where creatures switch between walk and run within a single spline chain. Not sure how client would even handle it.